packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -40,16 +40,16 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   FlutterMethodChannel *channel =
       [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/google_sign_in"
                                   binaryMessenger:[registrar messenger]];
-  FLTGoogleSignInPlugin *instance = [[FLTGoogleSignInPlugin alloc] init];
+  FLTGoogleSignInPlugin *instance = [[FLTGoogleSignInPlugin alloc] initWithController:[registrar messenger]];
```

Comment:
> initWithController:[registrar messenger]]

This is going to be a problem for you.  You are ignoring types and forcing a binary messenger to be a view controller?  Once you fix that, you'll find that the binary messenger isn't really a view controller anymore.

Reply:
Should I do something like this? 

```
    UIViewController *controller = [UIApplication sharedApplication].keyWindow.rootViewController;	
    [GIDSignIn sharedInstance].presentingViewController = controller;
```

Reply:
I don't know if grabbing the rootViewController here is going to work for you since the rootViewController might not still be the same rootViewController when GIDSignIn actually wants to present something.  Also, I'm not sure what it would do if the rootViewController was a UINavigationController and its topmost view is a FlutterViewController, would it display correctly?  You might have to try that out.

You might want to move the setter to the handle method call, is it possible GIDSignIn will want to display something without a method having triggered it?

Reply:
Also, i don't know if sending the FlutterViewController will work since I don't think there is a mechanism to do that?

Reply:
I agree that there probably isn't a mechanism to that.

I'm fine with grabbing the rootViewController at the time that the sign-in happens if you think that's a reasonable approach. It seems like the sign-in method is the only time that this gets used.

Reply:
I've update the approach as described, PTAL